src/services/job/create_index_for_big_table.service.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Action,
+  Service,
+} from '@ourparentcenter/moleculer-decorators-extended';
+import { Context, ServiceBroker } from 'moleculer';
+import BullableService, { QueueHandler } from '../../base/bullable.service';
+import { BULL_JOB_NAME, SERVICE } from '../../common';
+import knex from '../../common/utils/db_connection';
+
+@Service({
+  name: SERVICE.V1.JobService.CreateIndexForBigTable.key,
+  version: 1,
+})
+export default class CreateIndexForBigTableJob extends BullableService {
+  public constructor(public broker: ServiceBroker) {
+    super(broker);
+  }
+
+  public static buildQueryCreateIndex(payload: {
+    tableName: string;
+    indexName: string;
+    indexType: string;
+    columnName: string;
+    pagesPerRange: number;
+    whereClauses?: { column: string; expression: string; condition: string }[];
+  }): string {
+    let baseQuery = `
```

Comment:
dùng knex raw đi em, nó còn chạy sql escape

Reply:
knex.raw('CREATE INDEX CONCURRENTLY IF NOT EXISTS ${payload.indexName} ON ? USING ? ?', payload.tableName, ....)